---
title: "Final_Project"
author: "Prerna Sehgal"
date: "12/5/2019"
output: html_document
---
## R Markdown

## Description
Runx2 is a transcription factor that is overexpressed and promotes metastasis in breast cancer. In our pilot study we have shown that Runx2 undergoes post-translational modification. WT was found to promote TGF beta induced migration and invasion, but this ability was lost in case of the modification deficient mutant generated in the lab. The data suggests that this post-translational modification of Runx2 WT might promote metastasis. 

## Datasets
The ChIP-seq was done on 4 samples out of which 2 were inputs. I do not have replicates for the samples. The data was generated for 2 MDA-MB-231 breast cancer lines expressing Runx2 WT and Runx2 mutant and their respective inputs for normalization. I am starting with peaks data generated by MACS.

## Proposed Analysis
I want to look at binding pattern differences between the WT and mutant Runx2. 

For this I have four questions that I want to answer:

1. How is the enrichment of the protein binding affected across the chromosomes for WT and mutant?

2. Runx2 WT and mutant binding to TSS regions?

3. What regulatory elements are bound by the WT and the mutant?

4. Functional enrichment analysis for WT and mutant?

I will be using the ChIP-seq specific R Bioconductor packages to do my ChIP-seq analysis. 
```{r}
library(ChIPseeker)
library(ChIPpeakAnno)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene
library(clusterProfiler)
library(ReactomePA)
library(org.Hs.eg.db)
library(GenomicFeatures)
library(GenomicRanges)
library(ggplot2)
```

## Proposed Timeline & Major milestones (or segments).
Milestone 1 (11/14/2019): Get familiar with using the ChIPseeker package by working with a known data set. Try to get the result for the first two problems.

Milestone 2 (11/21/2019): Continue with understanding the package and answer the last two questions.

Milestone 3 (11/27/2019): Optimizing the results and overcome any roadblocks hit. Submit a rough draft for QA. 

Milestone 4 (12/06/2019): Address the QA problems and make the final draft. Submit the final project.

## Loading files
>- Loading MACS2 narrowpeaks data for Runx2 WT and the mutant files.
```{r}
setwd("~/desktop/files/peaks")
wt <- readPeakFile("IP-WT_S13_vs_Input-WT_S15_peaks.chrfiltered.bed")
wt
mutant <- readPeakFile("IP-4RK_S16_vs_Input-4RK_S14_peaks.chrfiltered.bed")
mutant
peaklist <- list(Runx2_WT = wt, Runx2_Mutant = mutant)
```

## ChIP Peaks Plot of WT and Mutant Runx2
>- This plot is used to know the peak locations over the whole genome.
>- Covplot function calculates the coverage of peak regions over chromosomes.
```{r fig.height=12, fig.width=12}
covplot(wt, weightCol="V5", title = "Runx2 WT ChIP Peaks over Chromosomes")
covplot(mutant, weightCol="V5", title = "Runx2 Mutant ChIP Peaks over Chromosomes")
```

>- There is reduction in the binding of Runx2 mutant over WT across the genome.

## Overlap of peaks between the WT and Mutant Runx2
>- To Test the overlaps of all the peaks. 
>- From the testing result, we can check the association between WT and Mutant Runx2. 
>- The vennDiagram shows the numbers of overlapping peak for these TFs.
```{r fig.cap="Overlap of peaks between the WT and Mutant Runx2", fig.align="center", fig.height=6, fig.width=6}
olpeaks <- findOverlapsOfPeaks(wt, mutant)
makeVennDiagram(olpeaks,
                fill=c("#009E73", "#F0E442"), # circle fill color
                col=c("#D55E00", "#0072B2"), #circle border color
                cat.col=c("#D55E00", "#0072B2"))
```

>- There are around 1000 peaks that overlap between the two transcription factors.

## Heatmap of ChIP binding of Runx2 WT and mutant to TSS regions
>- Calculating the profile of ChIP peaks binding to TSS regions.
>- To check if there is a difference between promoter binding pattern between the WT and mutant transcription factors.
```{r fig.cap="Heatmap of ChIP peaks binding to TSS regions", fig.align="center", fig.height=7, fig.width=5}
promoter <- getPromoters(TxDb = txdb, upstream=1000, downstream=1000)
tagMatrix <- lapply(peaklist, getTagMatrix, windows=promoter)
tagHeatmap(tagMatrix, xlim=c(-1000, 1000), color="red")
```

>- There is more binding of Runx2 mutant over Runx2 WT to promoter regions, suggesting it might activate certain genes not activated by the WT.

## Average profile of ChIP peaks binding of Runx2 WT and mutant to TSS regions
>- Compare the average binding profile of WT and Mutant Runx2.
>- Plot a graph to look at binding profile of the two transcription factors.
```{r fig.cap="Profile of ChIP peaks binding of Runx2 WT and mutant to TSS regions"}
plotAvgProf(tagMatrix, xlim=c(-1000, 1000), facet="row")
```

>- It shows the binding profile of Runx2 WT vs mutant.

## ChIP peak genomic annotation comparison between Runx2 WT and mutant.
>- This is done to annotate the location of a given peak in terms of genomic features. 
>- annotatePeak function assigns peaks to genomic annotation in “annotation”.
```{r}
peakAnnoList <- lapply(peaklist, annotatePeak, TxDb=txdb,
                       tssRegion=c(-1000, 1000), verbose=FALSE)
plotAnnoBar(peakAnnoList)
```

>- Runx2 WT binds more in intergenic and intronic regions which is expected as it is known to bind to enhancer regions.
>- Runx2 Mutant, on the other hand, binds to more promoter regions and shows less binding to intergenic and intronic regions. 
>- This reiterates the data shown above for binding of transcription factors around TSS regions.

## Visualize distribution of Runx2 WT and mutant binding loci relative to TSS
>- R function plotDistToTSS can be used to compare distance to TSS profiles among ChIPseq data.
>- plotDistToTSS can calculate the percentage of binding sites upstream and downstream from the TSS of the nearest genes, and visualize the distribution.
```{r}
plotDistToTSS(peakAnnoList)
```

>- This result recapitulates the point made by the previous result.
>- It shows higher binding near TSS by Runx2 mutant and higher binding for Runx2 WT in regions away from TSS.

## Functional Profile Comparison of Runx2 WT and mutant
>- The clusterProfiler package provides compareCluster function for comparing biological themes among gene clusters, and can be easily adopted to compare different ChIP peak experiments.
```{r fig.width=8, fig.height=6}
genes_list = lapply(peakAnnoList, function(i) as.data.frame(i)$geneId)
names(genes_list) = sub("_", "\n", names(genes_list))

comp_path <- compareCluster(geneClusters = genes_list, fun = "enrichPathway", pvalueCutoff  = 0.05, pAdjustMethod = "BH")                 
dotplot(comp_path, showCategory = 12, title = "Pathway Enrichment Analysis")
```

>- The plot below shows the different pathways in which the WT and mutant play a role. 
>- Some pathways are shared. 

## Overlap of annotated genes between Runx2 WT and mutant
>- vennplot function can be used to visualize the overlap of the nearest genes stored in peakAnnoList.
```{r fig.cap = "Overlap of annotated genes between Runx2 WT and mutant", fig.align = "center", fig.width=6, fig.height=8}
vennplot(genes_list)
```

>- The plot shows that Runx2 WT and mutant share binding sites around 7000 genes. 

## Data Mining with ChIP seq data deposited in GEO
>- There are many ChIP seq data sets that have been published and deposited in GEO database.
>- Compare my own dataset to those deposited in GEO to search for significant overlap data. 
>- Significant overlap of ChIP seq data by different binding proteins may be used to infer cooperative regulation and thus can be used to generate hypotheses.
```{r}
setwd("~/Desktop/files/peaks")
 
#Downloading GEO bed files and copying narrowpeaks file into another folder named "Narrowpeaks".

##downloadGEObedFiles(genome="hg38", destDir = "Geo_files")
##dir.create("~/Desktop/files/peaks/Narrowpeaks")
##listoffiles <- list.files(path = "~/Desktop/files/peaks/Narrowpeaks", pattern = ".narrowPeak.gz")
##listoffiles
##file.copy(listoffiles, "Narrowpeaks")

wt_overlap <- enrichPeakOverlap(queryPeak = wt, targetPeak = "Narrowpeaks", TxDb = txdb, pAdjustMethod = "BH", nShuffle = 1000, verbose = FALSE)
wt_report <- list()
wt_report[[1]] <- "Runx2 WT peak overlap enrichment analysis"
wt_report[[2]] <- wt_overlap
wt_report

mutant_overlap <- enrichPeakOverlap(queryPeak = mutant, targetPeak = "Narrowpeaks", TxDb = txdb, pAdjustMethod = "BH", nShuffle = 1000, verbose = FALSE)
mutant_report <- list()
mutant_report[[1]] <- "Runx2 mutant peak overlap enrichment analysis"
mutant_report[[2]] <- mutant_overlap
mutant_report
```